runtime/compiler/control/JITServerCompilationThread.cpp

```diff
@@ -84,6 +84,27 @@ computeServerMemoryState(TR::CompilationInfo *compInfo)
    return JITServer::ServerMemoryState::NORMAL;
    }
 
+/**
+ * @brief Helper method executed at the end of a compilation
+ * to determine whether the server is approaching the max. number of active threads.
+ */
+JITServer::ServerActiveThreadState
+computeServerActiveThreadState(TR::CompilationInfo *compInfo)
+   {
+   uint64_t highActiveThreadThreshold = 50;
+   uint64_t veryHighActiveThreadThreshold = 60;
```

Comment:
These two should probably come from -Xjit options so that we can apply some tuning at runtime should we have to.

Reply:
Ideally, the default values (when not specified via -Xjit options) for these thresholds should be computed based on the total number of threads. It could be something like `nthreads * 0.8` for high threshold and `nthreads - 4` for very high. @mpirvu What is your opinion on this?

Increasing the number of server threads from the current maximum of 63 is beneficial in certain scenarios, for instance when the network latency between the server and the clients is high, and having more in-flight requests (which need to be handled by more threads) improves performance. I will eventually submit a PR with changes to allow more threads on both the server and the client.

Reply:
Agreed. I would use 0.8 for high and 0.9 for very high, rounded to the nearest integer.